link.go

```diff
@@ -734,7 +734,7 @@ func (l *link) muxHandleFrame(fr frames.FrameBody) error {
 		//
 		// This isn't ideal, but there isn't a clear better way to handle it.
 		if fr, ok := fr.State.(*encoding.StateRejected); ok && errOnRejectDisposition {
-			return fr.Error
+			return &DetachError{fr.Error}
```

Comment:
Why is this a detach error? This is just a failed disposition, right?

Reply:
We have a setting, [LinkDetachOnDispositionError](https://github.com/Azure/go-amqp/blob/master/link_options.go#L364), that when enabled (true by default), says that disposition errors should detach the link.

Reply:
In fact, we should also return a `DetachError` from the first `Send` as well when this mode is enabled.

Reply:
It's a bit odd, but I think it makes sense.

The actual condition appears to only fire off if we're a _Sender_, which is different than I was thinking (had to actually look at the condition, nothing else appears to indicate that). It's interesting that it's true by default, but I think that's a discussion for another day.